lib/llvmopencl/WorkitemLoops.cc

```diff
@@ -93,6 +93,12 @@ WorkitemLoops::getAnalysisUsage(AnalysisUsage &AU) const
 
 }
 
+bool WorkitemLoops::doFinalization(llvm::Module &M) {
+  // todo what should it return
+  cleanParallelRegions();
```

Comment:
IIRC i tried deleting original_parallel_regions at the end of runOnFunction() or ProcessFunction(), but i got segfaults. I'm not familiar enough with the code to know where is it safe to delete, but this works (at least valgrind says so). Do you have a better place in mind?